### PR TITLE
[keycloak] add certificate, key and caCertificate to Route TLS configuration

### PR DESCRIPTION
**Pull request by secustor**
This PR add the possibility to add custom TLS keys and certifificates using the values.yml file.

### CODE REVIEW

File: charts/keycloak/values.yaml [hunk 1]
```diff
@@ -335,6 +335,21 @@ route:
     insecureEdgeTerminationPolicy: Redirect
     # TLS termination of the route. Can be `edge`, `passthrough`, or `reencrypt`
     termination: edge
+    # TLS custom certificate
+#   certificate: |
+#     aaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+#     bbbbbbbbbbbbbbbbbbbbbbbbbbbbb
+#     ccccccccccccccccccccccccccccc
```
Comment (mircohaug) [inline charts/keycloak/values.yaml:342]: Hi please use somewhat sane examples. eg
```
-----BEGIN CERTIFICATE-----
MIIDcjCCAlqgAwIBAgIUBTjKD8uJfpIN0uk9zsh+l5gALGcwDQYJKoZIhvcNAQEL
...
uWsAMNgqwe1Q9f3LitNXmsnmMPtpWg==
-----END CERTIFICATE-----
```

File: charts/keycloak/templates/route.yaml [hunk 2]
```diff
@@ -30,5 +30,17 @@ spec:
   tls:
     insecureEdgeTerminationPolicy: {{ $route.tls.insecureEdgeTerminationPolicy }}
     termination: {{ $route.tls.termination }}
+    {{- if $route.tls.certificate }}
+    certificate: |-
+{{ $route.tls.certificate | indent 6 }}
```
Comment (mircohaug) [inline charts/keycloak/templates/route.yaml:35]: Please use correct indentation and `nindent` instead

**Review by mircohaug — CHANGES_REQUESTED** [review]
Thanks for the PR. See the code comments for proposed changes

File: charts/keycloak/templates/route.yaml [hunk 3]
```diff
@@ -30,5 +30,17 @@ spec:
   tls:
     insecureEdgeTerminationPolicy: {{ $route.tls.insecureEdgeTerminationPolicy }}
     termination: {{ $route.tls.termination }}
+    {{- if $route.tls.certificate }}
```
Comment (mircohaug) [inline charts/keycloak/templates/route.yaml:33]: Sorry i forgot about this in the last review... Please also add the new values to the README

**Review by mircohaug — CHANGES_REQUESTED** [review]
One last little thing I forgot in the last review

**Comment by secustor** [comment]
@mircohaug Can you take another look at this?

File: charts/keycloak/README.md [hunk 4]
```diff
@@ -128,6 +128,9 @@ The following table lists the configurable parameters of the Keycloak chart and
 | `route.tls.enabled` | If `true`, TLS is enabled for the Route | `true` |
 | `route.tls.insecureEdgeTerminationPolicy` | Insecure edge termination policy of the Route. Can be `None`, `Redirect`, or `Allow` | `Redirect` |
 | `route.tls.termination` | TLS termination of the route. Can be `edge`, `passthrough`, or `reencrypt` | `edge` |
+| `route.tls.certificate` | Optionally tje TLS certificate of the route. | "" |
```
Comment (kofl91) [inline charts/keycloak/README.md:131]: Here is a small typo tje instead of the

**Review by kofl91 — DISMISSED** [review]
These changes make sense to me. And I could also use this. I have been manually adding a route.yaml to my keycloak deployments. This would help me get back to the upstream version of keycloak.

File: charts/keycloak/templates/route.yaml (hunk 3 above)
Comment (kofl91) [reply charts/keycloak/templates/route.yaml:33]: Can you mark this change as done? :) I would love to have this in the next Chart Release.

**Pull request closed without merging**